imap-types/src/core.rs

```diff
@@ -1475,12 +1469,34 @@ impl<T> IntoIterator for NonEmptyVec<T> {
     }
 }
 
-impl<T> AsRef<[T]> for NonEmptyVec<T> {
+impl<T, const N: usize> AsRef<[T]> for VecN<T, N> {
     fn as_ref(&self) -> &[T] {
         &self.0
     }
 }
 
+/// A `Vec` that always contains >= 1 elements.
+///
+/// Some messages in IMAP require a list of at least one element. We encoded these situations in a
+/// non-empty vector type to not produce invalid messages.
+///
+/// The `Debug` implementation equals `Vec` with an attached `+` at the end.
+pub type Vec1<T> = VecN<T, 1>;
+
+impl<T> From<T> for Vec1<T> {
+    fn from(value: T) -> Self {
+        VecN(vec![value])
+    }
+}
+
+/// A `Vec` that always contains >= 2 elements.
+///
+/// Some messages in IMAP require a list of at least one element. We encoded these situations in a
+/// non-empty vector type to not produce invalid messages.
+///
+/// The `Debug` implementation equals `Vec` with an attached `{2,}` at the end.
+pub type Vec2<T> = VecN<T, 2>;
```

Comment:
Hm, I'm not sure whether I like the name. Based on the name I'd assume that it has exactly 2 elements. Like the `Vec2` from the `glam` crate.

Alternatives:
- `Min2Vec`
- `Vec2Plus`
- `VecAtLeast2`
- `Vec2OrMoah`

Reply:
I see what you mean but still lean towards keeping `Vec2`. nom [uses](https://docs.rs/nom/latest/nom/multi/fn.separated_list1.html), e.g., `separated_list{0,1}` with the meaning >= 0 and >= 1. You may need to look it up, but given that I expect people to be often exposed to `Vec1`, maybe it's okay they need to look it up once? imap-types already has quite some "typing required" (and rightward drift). So -- while usually a bad argument! -- it makes a short name like `Vec2` hard to beat here... What do you think?

Reply:
Same for Vec1 and VecN. But no strong feelings from my side.